lib/modules/manager/maven/update.ts

```diff
@@ -78,7 +79,31 @@ export function bumpPackageVersion(
     const startTagPosition = versionNode.startTagPosition;
     const versionPosition = content.indexOf(versionNode.val, startTagPosition);
 
-    const newPomVersion = semver.inc(currentValue, bumpVersion);
+    let newPomVersion: string | null = null;
+    const currentPrereleaseValue = semver.prerelease(currentValue);
+    if (isSnapshot(currentPrereleaseValue)) {
+      // It is already a SNAPSHOT version.
+      // Therefore the same qualifier (prerelease) will be used as before.
+      let releaseType = bumpVersion;
+      if (!bumpVersion.startsWith('pre')) {
+        releaseType = ('pre' + bumpVersion) as ReleaseType;
+      }
+      newPomVersion = semver.inc(
+        currentValue,
+        releaseType,
+        currentPrereleaseValue?.join('.'),
```

Comment:
do you have a test where this is undefined?

Reply:
I think you mean `currentPrereleaseValue=undefined`:
It is impossible in this place that the value is undefined (if `isSnapshot()` returns true).
As the inverse can't be said, it's not possible to add a type predicate to `isSnapshot`.

I changed it to `currentPrereleaseValue!` to make this more clear.